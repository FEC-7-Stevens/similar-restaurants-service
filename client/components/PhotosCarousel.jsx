import React from 'react';

const PhotosCarousel = ({ photos }) => (
  <div>
    <img src="http://lorempixel.com/output/food-q-c-278-222-4.jpg" alt="" />
    {/* <img src={photos[0].url} alt="" /> */}
  </div>
);

export default PhotosCarousel;
