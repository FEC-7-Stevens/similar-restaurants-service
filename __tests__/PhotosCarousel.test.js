import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PhotosCarousel from '../client/components/PhotosCarousel';
import photosMocks from './../__mocks__/photos.mock.js';

configure({ adapter: new Adapter() });

describe('PhotosCarousel with photos data', () => {
  const photosCarouselProps = {
    photos: photosMocks.json()[0],
  };

  it('renders without crashing', () => {
    shallow(<PhotosCarousel {...photosCarouselProps} />);
  });

  it('renders main component correctly', () => {
    const carousel = mount(<PhotosCarousel {...photosCarouselProps} />);
    expect(carousel.exists('div')).toEqual(true);
  });

  it('PhotosCarousel renders snapshot correctly', () => {
    const component = renderer.create(<PhotosCarousel {...photosCarouselProps} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('PhotosCarousel without photos data', () => {
  it('renders main component correctly', () => {
    const carousel = mount(<PhotosCarousel photos={[{ url: 'https://zigat.s3-us-west-1.amazonaws.com/no-dish.png' }]} />);
    expect(carousel.exists('div')).toEqual(true);
  });
});
