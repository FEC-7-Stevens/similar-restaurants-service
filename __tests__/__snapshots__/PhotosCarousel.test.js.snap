// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotosCarousel with photos data PhotosCarousel renders snapshot correctly 1`] = `
<div
  className="PhotosCarousel__Carousel-sc-158497q-0 gfVLiU"
>
  <img
    alt="restaurant photo"
    className="PhotosCarousel__Slide-sc-158497q-1 koTQkm"
    src="https://zigat.s3-us-west-1.amazonaws.com/image8.jpg"
  />
  <button
    className="PhotosCarousel__Button-sc-158497q-2 cMeYdq btn-left"
    onClick={[Function]}
  >
    &lt;
  </button>
  <button
    className="PhotosCarousel__Button-sc-158497q-2 cMeYdq btn-right"
    onClick={[Function]}
  >
    &gt;
  </button>
</div>
`;
